scripts/svm/executeRebalanceToHubPool.ts

```diff
@@ -0,0 +1,482 @@
+/**
+ * Script: Execute USDC Rebalance to Hub Pool
+ *
+ * This script executes a previously proposed root bundle on the Hub Pool
+ * to rebalance USDC from the Solana Spoke Pool to the Ethereum Hub Pool.
+ * It handles CCTP attestations, relayer refund leaf execution, and prepares
+ * pending liabilities for bridging back to the Hub Pool.
+ *
+ * Required Environment Variables:
+ * - TESTNET: (Optional) Set to "true" to use Sepolia; defaults to mainnet.
+ * - MNEMONIC: Wallet mnemonic to sign the Ethereum transaction.
+ * - HUB_POOL_ADDRESS: Ethereum address of the Hub Pool.
+ * - NODE_URL_1: Ethereum RPC URL for mainnet (ignored if TESTNET=true).
+ * - NODE_URL_11155111: Ethereum RPC URL for Sepolia (ignored if TESTNET=false).
+ *
+ * Required Arguments:
+ * - `--netSendAmount`: The unscaled amount of USDC to rebalance (e.g., for USDC with 6 decimals, 1 = 0.000001 USDC).
+ * - `--resumeRemoteTx`: (Optional) Hash of a previously submitted remote transaction to resume.
+ *
+ * Example Usage:
+ * TESTNET=true \
+ * NODE_URL_11155111=$NODE_URL_11155111 \
+ * MNEMONIC=$MNEMONIC \
+ * HUB_POOL_ADDRESS=$HUB_POOL_ADDRESS \
+ * anchor run executeRebalanceToHubPool \
+ * --provider.cluster "devnet" \
+ * --provider.wallet $SOLANA_PKEY_PATH \
+ * -- --netSendAmount 7
```

Comment:
so this is provided to re-construct the merkle leaf & proof? it also assumes that there are no relayer repayments within the leaf, right?

in practice, I don't think this is how this flow will work, right? it's assumed that the data worker will post the leafs & proofs to a public location so these can be retrieved. this particular script flow is only useful in this particular situation where you don't have any relayer repayments & we only do a net repayment.

It would be useful to be able to pass in the leafs as en env/json or something like this? this would let you take them from the script that does the proposing, save them, and then not need to re-generate them. it would also better track with what things will look like when we interface with the data worker later (executor should not need to generate leafs, just get given them)

Reply:
(prehaps this is still useful if we are dealing with JUST creating leafs to pull/push tokens to/from SVM via the netSendAmount

Reply:
Yes, exactly. The `netSendAmount` is merely to avoid having to pass leaves, roots, proofs, etc., between this script and `proposeRebalanceToHubPool.ts`. It also assumes that the `poolRebalanceTree` has a single leaf with empty data and the solanaChainId. Thus, both scripts are dependent on each other. I believe they do not showcase a real production scenario; they are more like helpers to test the rebalance from the SVM Spoke to the HubPool.

On the other hand, I separated `bridgeLiabilityToHubPool.ts` from them because this one could represent a real use case, as you pointed out. It’s essentially what the finalizer needs to do.